cobalt/browser/h5vcc_runtime/deep_link_manager.cc

```diff
@@ -49,5 +49,28 @@ const std::string DeepLinkManager::GetAndClearDeepLink() {
   return value;
 }
 
+void DeepLinkManager::AddListener(
+    mojo::Remote<DeepLinkListener> listener_remote) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  // mojo::RemoteSet removes the listener_remote automatically when the Mojo
+  // client disconnects
+  listeners_.Add(std::move(listener_remote));
+}
+
+void DeepLinkManager::OnDeepLink(const std::string& url) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  if (listeners_.empty()) {
+    // Deeplink is held until a callback is registered, at which point it will
+    // be consumed."
+    set_deep_link(url);
+  } else {
+    // No need to worry about race condition because all access to the
+    // mojo::RemoteSet happens on one thread.
+    for (auto& listener : listeners_) {
```

Comment:
Question: On Chrobalt, is it ever actually happening that we have more than one `listeners_`?
- If yes: then this code is OK
- If not: plz remove `listeners_` and instead just have a single `mojo::Remote<DeepLinkListener> listener_` in this class, then leave in `AddListener()` a massive `CHECK(!listener_)`;` to enforce that there's never more than one listener registered. If that CHECK is ever hit, it means the assumption was wrong and we should support multiple listeners, and we'd be in the case above ("If yes"), otherwise we have a simpler code : )

Reply:
Gotcha, we need it and we are going to use it presently, so let's keep it.